### ISSUE TITLE
Most keyboard shortcuts don't work at all (chrome)

### ISSUE THREAD

**Issue by basementRot** [gorhill/uMatrix#678]
Shortcuts can be set in Chrome options under Extensions > Keyboard Shortcuts (at bottom).

These are all broken...
"Open dashboard"
"Revert all temporary changes"
"Temporarily whitelist all" 
"Temporarily whitelist page domain"

"Activate the Extension" shortcut actually works. It makes the uMatrix window popup.

I've tried setting the shortcuts to various key combos of ctrl, alt and/or shift + letters. I've also tried setting the shortcut option to Global.

I'm new to uMatrix and really miss being able to hit shift+z to whitelist pages. Shortcut keys are one of the most important usability features in my opinion. 

Windows 10, Chrome v 54.0.2840.99 m (64-bit)

**Comment by basementRot** [comment]
I should also add that other people have claimed to have this problem too. Nobody had made a specific bug report about it though.

**Comment by jonas2515** [comment]
Yeah, I have the same problem, but no keyboard shortcuts are working at all. I'm using Chromium 55.0.2883.75 and uMatrix 0.9.3.6 on Debian unstable.

**Comment by AdKiller** [comment]
praying for the day uMatrix finally gets keyboard shortcuts... I am willing to pay for its development.

**Comment by IgelRM** [comment]
I can confirm this too, using Chrome v56.0.2924.87 and uMatrix v0.9.3.6.
@gorhill can you add a word on this please. I am using uMatrix since quite some time and liking it.

**Comment by gorhill** [comment]
Adding keywords to uMatrix Chrome means I would have to retrofit a new API in uMatrix Firefox -- which is a pain and really just do not have the time to deal with. Once uMatrix is a proper Firefox WebExtensions, everything will be so much easier to deal with time-wise and effort-wise, as their won't be two different code bases to maintain. In short, keyboard shortcuts will be back once uMatrix is Firefox WebExtensions-ready.

**Comment by madduck** [comment]
AFAICT, uMatrix is now a proper WebExtension, and I also just came here longing for key combos to control uMatrix.

Any news?

**Comment by gorhill** [comment]
There is no UI yet in Firefox to assign keyboard shortcuts.

**Comment by IgelRM** [comment]
Could you set up shortcuts for Chrome with the WebExtension standard as they should also function with Firefox when a UI later gets implemented, if I understand this correctly.

**Comment by gorhill** [comment]
The old shortcuts that were still there were meant for HTTPSB, and with the refactoring, a lot of these shortcuts lo longer made sense. So I am back at deciding which shortcuts are good to have for uMatrix. I don't want a kitchen sink approach, only those shortcuts which make a lot of sense should be available.

**Comment by madduck** [comment]
I'm envisioning being able to hit e.g. ctrl-alt-m to have the dialog pop open, using the arrow keys to move around like in a spreadsheet and then the ability to toggle the verdict for columns, rows, and single cells with e.g. enter.

Once you have that, then adding additional shortcuts should be simple, maybe even macros.

**Comment by basementRot** [comment]
Well I click "Disable/enable matrix filtering for this scope" about a million times a day, so a hotkey (like shift+z) would be fantastic. This may not be the best, security wise, but uMatrix seems to break vital features on a great many websites. And on the well known, probably safe sites I really don't feel like taking the time to be choosy, carefully selecting all my blocks/allows.

Maybe you can think of some good hotkeys to solve what I see as the main problem.
uMatrix blocks important scripts, this ends up breaking a websites important functionality. Things like logging in to a site, or making purchases with shopping carts. It would be nice to fix problems like that with a simple hotkey, especially on most big sites that people trust.
